doc/developers/advanced_installation.rst

```diff
@@ -0,0 +1,427 @@
+
+Replace ``v7.0`` by the ``v7.1`` in the above commands to do the same for
+Python 3 instead of Python 2.
+
+Replace ``/x64`` by ``/x86``  to build for 32-bit Python instead of 64-bit
+Python.
+
+
+Building binary packages and installers
+---------------------------------------
+
+The ``.whl`` package and ``.exe`` installers can be built with::
+
+    pip install wheel
+    python setup.py bdist_wheel bdist_wininst -b doc/logos/scikit-learn-logo.bmp
+
+The resulting packages are generated in the ``dist/`` folder.
+
+
+Using an alternative compiler
+-----------------------------
+
+It is possible to use `MinGW <http://www.mingw.org>`_ (a port of GCC to Windows
+OS) as an alternative to MSVC for 32-bit Python. Not that extensions built with
+mingw32 can be redistributed as reusable packages as they depend on GCC runtime
+libraries typically not installed on end-users environment.
+
+To force the use of a particular compiler, pass the ``--compiler`` flag to the
+build step::
+
+    python setup.py build --compiler=my_compiler install
+
+where ``my_compiler`` should be one of ``mingw32`` or ``msvc``.
+
+
+.. _install_bleeding_edge:
+
+Bleeding Edge
+=============
+
+See section :ref:`git_repo` on how to get the development version. Then follow
+the previous instructions to build from source depending on your platform.
+
+
+.. _testing:
+
+Testing
+=======
+
+Testing scikit-learn once installed
+-----------------------------------
+
+Testing requires having the `nose
+<http://somethingaboutorange.com/mrl/projects/nose/>`_ library. After
+installation, the package can be tested by executing *from outside* the
+source directory::
+
+    $ nosetests -v sklearn
+
+Under Windows, it is recommended to use the following command (adjust the path
+to the ``python.exe`` program) as using the ``nosetests.exe`` program can badly
+interact with tests that use ``multiprocessing``::
+
+    C:\Python34\python.exe -c "import nose; nose.main()" -v sklearn
+
+This should give you a lot of output (and some warnings) but
+eventually should finish with a message similar to::
+
+    Ran 3246 tests in 260.618s
+    OK (SKIP=20)
+
+Otherwise, please consider posting an issue into the `bug tracker
+<https://github.com/scikit-learn/scikit-learn/issues>`_ or to the
+:ref:`mailing_lists` including the traceback of the individual failures
+and errors.
+
+
+Testing scikit-learn from within the source folder
+--------------------------------------------------
+
+Scikit-learn can also be tested without having the package
+installed. For this you must compile the sources inplace from the
+source directory::
+
+    python setup.py build_ext --inplace
+
+Test can now be run using nosetests::
+
+    nosetests -v sklearn/
+
+This is automated by the commands::
+
+    make in
+
+and::
+
+    make test
+
+
+You can also install a symlink named ``site-packages/scikit-learn.egg-link``
+to the development folder of scikit-learn with::
+
+    pip install --editable .
```

Comment:
Not sure which one is the recommended one, but we could also mention `python setup.py develop`